nixos/doc/manual/development/running-nixos-tests-interactively.section.md

```diff
@@ -38,6 +38,12 @@ directory.
 
 ## Interactive-only test configuration {#sec-nixos-test-interactive-configuration}
 
-You can add configuration that is specific to the interactive test driver, by adding to the `interactive` option.
-`interactive` is a copy of the regular test options namespace, and is used by the interactive test driver.
-It can be helpful for troubleshooting changes that you don't want to apply to regular test runs.
+The `.driverInteractive` attribute combines the regular test configuration with
+definitions from the [`interactive` submodule](#opt-interactive). This gives you
+a more usable, graphical, but slightly different configuration.
+
+To run only the regular configuration, build the `<test>.driver` attribute
+instead, and call it with the flag `result/bin/nixos-test-driver --interactive`.
+
+You can add your own interactive-only test configuration by adding extra
+configuration to the [`interactive` submodule](#opt-interactive).
```

Comment:
Not sure if I read the code correctly, but `driverInteractive` seems to be only used internally. If that's true it should not even be mentioned here.

```suggestion
You can add your own interactive-only test configuration by adding extra
configuration to the [`interactive` submodule](#opt-interactive).

It will be combined with the regular test configuration, and provides a more convenient, graphical test environment.

To run interactvely the regular configuration, build the `<test>.driver` attribute
instead, and call it with the `interactive` flag. For example:

    result/bin/nixos-test-driver --interactive


```

Reply:
`.driverInteractive` is specifically for troubleshooting the user's tests, so it is a public interface.

So there's three ways to run a test
 - regular configuration, non-interactive
 - regular configuration, interactive
 - interactive configuration, interactive

I considered adding this table, but I find the cell contents too vague.

|                                           | regular mode              | `--interactive` mode                                      |
|-------------------------------------------|---------------------------|-----------------------------------------------------------|
| regular configuration                     | `nix-build --no-out-link` | `nix-build -A nixosTests.<name>.driver` + `--interactive` |
| graphical and `interactive` configuration | not useful                | `nix-build -A nixosTests.<name>.driverInteractive`        |

The underlying problem is that we have two uses of "interactive" that aren't quite the same.
 - interactive configuration: a graphical qemu among other things
 - running the driver interactively: with `--interactive`, which is also implied by `.driverInteractive`.

I'd like for this to be improved, but this is all pre-existing and I'll consider it out of scope for this PR.

Reply:
> So there's three ways to run a test

This was clear to me from the description. What's not at all clear how `driverInteractive` plays into this and how one would, for example as you say, troubleshoot setups.

Reply:
You'd follow the instructions and then you get a python repl where you can do anything the `testScript` can do, as described in `#sec-running-nixos-tests-interactively`.

> how driverInteractive plays into this

It seems that this paragraph answers your question:

> The `.driverInteractive` attribute combines the regular test configuration with
> definitions from the [`interactive` submodule](#opt-interactive). This gives you
> a more usable, graphical, but slightly different configuration.

Could you perhaps refine your question?